numba/cuda/tests/cudapy/test_optimization.py

```diff
@@ -0,0 +1,82 @@
+import numpy as np
+
+from numba.cuda.testing import skip_on_cudasim, CUDATestCase
+from numba import cuda, float64
+import unittest
+
+
+def kernel_func(x):
+    x[0] = 1
+
+
+def device_func(x, y, z):
+    return x * y + z
+
+
+# Fragments of code that are removed from kernel_func's PTX when optimization
+# is on
+removed_by_opt = ( '__local_depot0', 'call.uni', 'st.param.b64')
+
+
+@skip_on_cudasim('Simulator does not optimize code')
+class TestOptimization(CUDATestCase):
+    def test_eager_opt(self):
+        # Optimization should occur by default
+        kernel = cuda.jit((float64[::1],))(kernel_func)
+        ptx = kernel.inspect_asm()
+
+        for fragment in removed_by_opt:
+            with self.subTest(fragment=fragment):
+                self.assertNotIn(fragment, ptx)
+
+    def test_eager_noopt(self):
+        # Optimization disabled
+        kernel = cuda.jit((float64[::1],), opt=False)(kernel_func)
+        ptx = kernel.inspect_asm()
+
+        for fragment in removed_by_opt:
+            with self.subTest(fragment=fragment):
+                self.assertIn(fragment, ptx)
+
+    def test_lazy_opt(self):
+        # Optimization should occur by default
```

Comment:
Isn't it being spelled out in the decorator on the next line?

Reply:
Yes, I should not have the `opt` kwarg in there.

Reply:
I've now removed the `opt` kwarg on the next line.

Reply:
thanks.